syft/frameworks/torch/linalg/lr.py

```diff
@@ -284,3 +285,247 @@ def _compute_pvalues(self):
+
+        # Secred share tensors between hbc_worker, crypto_provider and a random worker
+        # and compute aggregates. It corresponds to the Combine stage of DASH's algorithm
+        idx = random.randint(0, len(self.workers) - 1)
+        XX_shared = sum(self._share_ptrs(XX_ptrs, idx)).sum(dim=0)
+        Xy_shared = sum(self._share_ptrs(Xy_ptrs, idx))
+        yy_shared = sum(self._share_ptrs(yy_ptrs, idx))
+        CX_shared = sum(self._share_ptrs(CX_ptrs, idx))
+        Cy_shared = sum(self._share_ptrs(Cy_ptrs, idx))
+        R_shared = torch.cat(self._share_ptrs(R_ptrs, idx), dim=0)
+
+        # QR decomposition of R_shared
+        _, R = qr(R_shared, norm_factor=self.total_size ** (1 / 2))
+
+        # Compute inverse of upper matrix
+        R_inv = self._inv_upper(R)
+
+        Qy = R_inv.t() @ Cy_shared
+        QX = R_inv.t() @ CX_shared
+
+        denominator = XX_shared - (QX ** 2).sum(dim=0)
+        # Need the line below to perform inverse of a number in MPC
+        inv_denominator = ((0 * denominator + 1) / denominator).squeeze()
+
+        coef_shared = (Xy_shared - QX.t() @ Qy).squeeze() * inv_denominator
+
+        sigma2_shared = (
+            (yy_shared - Qy.t() @ Qy).squeeze() * inv_denominator - coef_shared ** 2
+        ) / self._dgf
+
+        self.coef = coef_shared.get().float_precision()
+        self.sigma2 = sigma2_shared.get().float_precision()
+        self.se = self.sigma2 ** (1 / 2)
+
+        self._compute_pvalues()
+
+    def get_coeff(self):
+        return self.coef
+
+    def get_standard_errors(self):
+        return self.se
+
+    def get_p_values(self):
+        return self.pvalue
+
+    def _check_ptrs(self, X_ptrs, C_ptrs, y_ptrs):
+        """
+        Method that check if the lists of pointers corresponding to the response vector,
+        transient covariate vectors and independent permanent covariate vectors have
+        their elements as expected. It also computes parallelly some Regressor's
+        attributes such as degrees of freedom and total sample size.
+        """
+        # Set number of features
+        self.n_features = X_ptrs[0].shape[1]
+        # Set number of permanent covariate features
+        self.n_permanent = C_ptrs[0].shape[1]
+
+        x_size, c_size, y_size = 0, 0, 0
+        for x, c, y in zip(X_ptrs, C_ptrs, y_ptrs):
+            # Check wrapper
+            if not (x.has_child() and c.has_child() and y.has_child()):
+                raise TypeError(
+                    "Some tensors are not wrapped, please provide a wrapped Pointer Tensor"
+                )
+
+            # Check if x, c and y are pointers
+            if not (
+                isinstance(x.child, PointerTensor)
+                and isinstance(c.child, PointerTensor)
+                and isinstance(y.child, PointerTensor)
+            ):
+                raise TypeError(
+                    "Some tensors are not pointers, please provided a wrapped Pointer Tensor"
```

Comment:
Can you fuse this if statement and the previous together (to make the code shorter)
`all(map(lambda t: t.has_child() and isinstance(t.child, PointerTensor), (x, c, y)))`